tests/engine/adhoc_test.go

```diff
@@ -170,3 +172,28 @@ func recursiveEntry(t *testing.T, newEngine func() wasm.Engine) {
 	_, _, err = store.CallFunction("test", "main", uint64(1))
 	require.NoError(t, err)
 }
+
+func importedAndExportedFunc(t *testing.T, newEngine func() wasm.Engine) {
+	buf, err := os.ReadFile("testdata/imported_and_exported_func.wat")
+	require.NoError(t, err)
+	mod, err := text.DecodeModule(buf)
+	require.NoError(t, err)
+
+	store := wasm.NewStore(newEngine())
+
+	// addInt is imported by the wasm module, and the wasm module exports addInt back
```

Comment:
ex if you inline you will already have the comments about what's going on in the source in the same test